### PR TITLE
[TF generation] Fix typo

### PR DESCRIPTION
**Pull request by SidJain1412**
# What does this PR do?

Typo + Parameter Assertion Fix

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dimiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests), 
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to the it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by SidJain1412** [comment]
I see that tests are failing, but shouldn't `min_length` and `top_k` simply not be allowed to go to zero?

**Comment by patrickvonplaten** [comment]
`min_length` defaults to 0 which is expteced behavior. `num_beams` can also be 0. `top_k` is 0 if it is not used => so I don't think we should do these changes.

File: src/transformers/generation_tf_utils.py [hunk 1]
```diff
@@ -221,13 +221,13 @@ def generate(
...
-        assert isinstance(min_length, int) and min_length >= 0, "`min_length` should be a positive integer."
+        assert isinstance(min_length, int) and min_length > 0, "`min_length` should be a positive integer."
         assert isinstance(do_sample, bool), "`do_sample` should be a boolean."
         assert isinstance(early_stopping, bool), "`early_stopping` should be a boolean."
         assert isinstance(use_cache, bool), "`use_cache` should be a boolean."
-        assert isinstance(num_beams, int) and num_beams > 0, "`num_beams` should be a strictely positive integer."
+        assert isinstance(num_beams, int) and num_beams > 0, "`num_beams` should be a strictly positive integer."
         assert temperature > 0, "`temperature` should be strictely positive."
-        assert isinstance(top_k, int) and top_k >= 0, "`top_k` should be a positive integer."
+        assert isinstance(top_k, int) and top_k > 0, "`top_k` should be a positive integer."
```
Comment (patrickvonplaten) [inline src/transformers/generation_tf_utils.py:230]:
```suggestion
        assert isinstance(top_k, int) and top_k >= 0, "`top_k` should be a positive integer."
```

File: src/transformers/generation_tf_utils.py [hunk 2]
```diff
@@ -221,13 +221,13 @@ def generate(
             batch_size = 1
 
         assert isinstance(max_length, int) and max_length > 0, "`max_length` should be a strictely positive integer."
-        assert isinstance(min_length, int) and min_length >= 0, "`min_length` should be a positive integer."
+        assert isinstance(min_length, int) and min_length > 0, "`min_length` should be a positive integer."
```
Comment (patrickvonplaten) [inline src/transformers/generation_tf_utils.py:224]:
```suggestion
        assert isinstance(min_length, int) and min_length >= 0, "`min_length` should be a positive integer."
```

**Comment by patrickvonplaten** [comment]
We can fix the typo though ;-)

**Comment by SidJain1412** [comment]
My bad, that makes sense 😄

**Pull request merged**